actions/treasures.go

```diff
@@ -15,7 +24,51 @@ type treasureReq struct {
 	EthKey          string `json:"EthKey"`
 }
 
+type treasureRes struct {
+	Success string `json:"success"`
+}
+
+func unixMilli(t time.Time) int64 {
+	return t.Round(time.Millisecond).UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
-	return nil
-}
+	req := treasureReq{}
+	oyster_utils.ParseReqBody(c.Request(), &req)
+
+	host_ip := os.Getenv("HOST_IP")
+	provider := "http://" + host_ip + ":14265"
+	api := giota.NewAPI(provider, nil)
+	ftr := &giota.FindTransactionsRequest{Bundles: []giota.Trytes{"DEXRPLKGBROUQMKCLMRPG9HFKCACDZ9AB9HOJQWERTYWERJNOYLW9PKLOGDUPC9DLGSUH9UHSKJOASJRU"}}
+	resp, err := api.FindTransactions(ftr)
+
+	datamap1, vErr := models.GetDataMap(req.GenesisHash, req.NumChunks)
+	for _, d := range datamap1 {
+		
+	}
+	// or 
+	
+	vErr, error := models.BuildDataMaps(req.GenesisHash, req.NumChunks)
+	datamap, err := models.GetDataMapByGenesisHashAndChunkIdx(req.GenesisHash, req.NumChunks)
```

Comment:
This method queries the DB data_maps table.  The data_map that the brokernode builds for claiming the treasure shouldn't end up in the data_maps table.

Reply:
Ok, so the same like previous comment... should I save it into db or on fly? Or dont save?

Reply:
Do it on the fly.  After mainnet launch we'll explore moving it to a table.